IPython/core/guarded_eval.py

```diff
@@ -927,6 +960,25 @@ def dummy_function(*args, **kwargs):
     if isinstance(node, ast.BinOp):
         left = eval_node(node.left, context)
         right = eval_node(node.right, context)
+        if is_type_annotation(left) and is_type_annotation(right):
```

Comment:
If I recall correctly that type annotations only support `|` binary operation then we can reject more quickly here:


```suggestion
        if node.op is ast.BitOr and is_type_annotation(left) and is_type_annotation(right):
```